### ISSUE TITLE
Fix false positives for each functions

### ISSUE THREAD

**Issue by AoDev** [stylelint/stylelint#4022]
> Clearly describe the bug

In `less`, I can generate classes using [the each function](http://lesscss.org/functions/#list-functions-each), example:

```
@container-values: 64em, 24em;

each(@container-values, .(@cvalue) {
  .container-@{cvalue} {
    max-width: @cvalue;
    margin-left: auto;
    margin-right: auto;
  }
})
```

It seems that stylelint can not process this correctly, I've got a serie of issues:
- `identation` (it asks me to put it at 0).
- `Unexpected "" (selector-descendant-combinator-no-non-space)stylelint(selector-descendant-combinator-no-non-space)`
- `Expected no more than 0 empty lines (selector-max-empty-lines)stylelint(selector-max-empty-lines)`

I have tried to rewrite this in multiple ways to please stylelint but I could not.

> What stylelint configuration is needed to reproduce the bug?

I am using 
```
{
  "extends": "stylelint-config-standard"
}

```
> Which version of stylelint are you using?

```
    "stylelint": "9.10.1",
    "stylelint-config-standard": "18.2.0",
```
> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

npm script: `"lint:css": "stylelint \"src/**/*.less\"",`

I don't know what the right style is but I took example from lesscss.org docs.

**Comment by evilebottnawi** [comment]
I think it is bug in less parser again. It's always wrong with him.

**Comment by AoDev** [comment]
A secondary problem is that I can not skip it, since I don't have many of these it would be ok to just disable stylelint.
But surrounding the code above with
```
/* stylelint-disable */
[code]
/* stylelint-enable */
```
does not work. Stylelint is still disabled and does not obey the `/* stylelint-enable */`part.

**Comment by jeddy3** [comment]
@AoDev Thanks for the report and for using the template.

However, I'm not able to reproduce this. For me there are no violations for stylelint-config-standard. This makes sense because the postcss-less seems to parse `each` functions as at-rules. For example, given:

```less
each(@selectors, {
  .sel-@{value} {
    a: b;
  }
});
```

The postcss-less produced AST looks like:

```
nodes: Array (1 item)
  0: AtRule
    name: "each"
    params: "(@selectors, {\n  .sel-@{value} {\n    a: b;\n  }\n}) 
```

The rules built into stylelint are geared towards standard syntax and ignore the `each` construct.

Please update to stylelint@10 and see what happens. If the issue persists, can you create a reproducible test repo so we can help further?

**Comment by AoDev** [comment]
@jeddy3 Thanks for the answer. Actually I think I'm on a really weird case here. 
I was preparing a repo, then couldn't reproduce the issue... very confusing. 
Then I played around and I discovered that I have this problem only if there are other rules in the less file.
I never thought about that so, in my example above, I only copied the "each" part.
But, if I add another rule in the same less file, then I get the issue.

This won't cause an issue:
```less
// SPACERS
// generates classes like space-top-1, space-bottom-2, space-left-3, ...
@spacer-directions: top, right, bottom, left;
@spacer-values: 0, 1, 2, 3, 4;

each(@spacer-values, .(@svalue) {
  each(@spacer-directions, .(@direction) {
    .space-@{direction}-@{svalue} {
      margin-@{direction}: @spacing-base * @svalue;
    }
  })
})
```

But this will:
```less
// SPACERS
// generates classes like space-top-1, space-bottom-2, space-left-3, ...
@spacer-directions: top, right, bottom, left;
@spacer-values: 0, 1, 2, 3, 4;

each(@spacer-values, .(@svalue) {
  each(@spacer-directions, .(@direction) {
    .space-@{direction}-@{svalue} {
      margin-@{direction}: @spacing-base * @svalue;
    }
  })
})

.padded-horizontal-2 {
  padding-left: @spacing-base * 2;
  padding-right: @spacing-base * 2;
}

```

Results in this:
```
  7:3  ✖  Expected indentation of 0        indentation
          spaces
  8:5  ✖  Expected indentation of 2        indentation
          spaces
  9:7  ✖  Expected indentation of 4        indentation
          spaces
 10:5  ✖  Expected indentation of 2        indentation
          spaces
 11:3  ✖  Expected indentation of 0        indentation
          spaces
 12:3  ✖  Unexpected " "                   selector-descendant-combinator-no-non-space
 12:3  ✖  Expected no more than 0 empty    selector-max-empty-lines
          lines
```

I can still publish the repo if you want me to.

**Comment by jeddy3** [comment]
@AoDev Thanks for digging deeper.

I can reproduce the violations with:

```less
each(@selectors, {
  .sel-@{value} {
    a: b;
  }
})

a {}
```

However, are you sure that is valid Less syntax?

The `each` examples in the [Less docs](http://lesscss.org/functions/#list-functions-each) are all terminated with a semicolon. The following (note the terminating semicolon) produces no violations:

```less
each(@selectors, {
  .sel-@{value} {
    a: b;
  }
});

a {}
```

If you're sure that your syntax is valid then we should raise an issue upstream in postcss-less. If you're not, then add a terminating semicolon and everything should hopefully be OK.

**Comment by AoDev** [comment]
@jeddy3 I can confirm that adding semi-colons remove the violations errors, but my code does compile correctly without them. The expected CSS classes are generated and can be used in my web apps without problem. 

It feels a bit like JS semicolons, not needed most of the time but we have linting rules if we want to enforce a style. For less.js, I don't know, since it compiles, I'm assuming it's valid syntax with or without the semicolon.
On my side I'll just add them in my code as a fix so I'm not blocked, but I am fine with both style.

**Comment by jeddy3** [comment]
I'll close for now. We can always reopen if this issue crops up again.
